wp-content/themes/core/blocks/tribe/terms/render.php

```diff
@@ -15,45 +15,35 @@
 
 echo '<div ' .  wp_kses_data( get_block_wrapper_attributes() ) . '>';
 
-if ( 0 === count( $terms_block_terms ) ) {
-	echo '<!-- Terms block: No terms to list. -->';
-	echo '</div>';
-
-	return;
+// No terms and we're in the block editor
+if ( 0 === count( $terms_block_terms ) && ( defined( 'REST_REQUEST' ) && REST_REQUEST ) ) {
+	echo '<span class="wp-block-tribe-terms__empty-msg t-category">';
+	printf( esc_html__( "No %s assigned to post", 'tribe' ), $terms_block->get_taxonomy_name() );
+	echo '</span>';
 }
 
-if ( count( $terms_block_terms ) > 1 ) {
-	echo  '<ul class="wp-block-tribe-terms__list">';
-}
+if ( count( $terms_block_terms ) > 0 ) {
+	echo '<ul class="wp-block-tribe-terms__list">';
 
-foreach ( $terms_block_terms as $term ) {
-	if ( count( $terms_block_terms ) > 1 ) {
```

Comment:
I don't think it's worth switching the markup depending on if there is only a single term displayed. The block itself is a "terms block" which infers there could be one or more.